x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.mapper;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.TextSearchInfo;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.script.Script;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.search.SearchService.ALLOW_EXPENSIVE_QUERIES;
+
+/**
+ * Abstract base {@linkplain MappedFieldType} for scripted fields.
+ */
+abstract class AbstractScriptMappedFieldType extends MappedFieldType {
+    protected final Script script;
+
+    AbstractScriptMappedFieldType(String name, Script script, Map<String, String> meta) {
+        super(name, false, false, TextSearchInfo.NONE, meta);
+        this.script = script;
+    }
+
+    @Override
+    public final String typeName() {
+        return ScriptFieldMapper.CONTENT_TYPE;
+    }
```

Comment:
shall we add some way to not forget about overriding familyType for all the subclasses? Maybe we could override the method and make it call some other new abstract method?

Reply:
oh well I see below that you have runtimeType, so it would be enough to override familyTypeName and call runtimeType in there.

Reply:
Got it!